### ISSUE TITLE
@Searchable generated resolver are unable to return data & IAM is unauthorized for all queries.

### ISSUE THREAD

**Issue by Rafcin** [aws-amplify/amplify-cli#9424]
### Before opening, please confirm:

- [X] I have installed the latest version of the Amplify CLI (see above), and confirmed that the issue still persists.
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-cli/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### How did you install the Amplify CLI?

npm install -g @aws-amplify/cli

### If applicable, what version of Node.js are you using?

14.18.1

### Amplify CLI Version

7.6.5

### What operating system are you using?

Ubuntu 20

### Amplify Categories

auth, api

### Amplify Commands

Not applicable

### Describe the bug

I'm looking in Appsync at my resolvers testing if everything works after the migration, and it looks like the resolvers that are generated for models with the searchable directive no longer function properly. When running searchable resolvers using API key they only return 0 results. In addition any query I attempt to do with IAM returns an unauthorized error.

### Expected behavior

The searchable queries should return some values and all queries should be accessible by IAM as they had been.

### Reproduction steps

Running migration as instructed.

### GraphQL schema(s)

<details>

```graphql
enum ModelStatus {
  DRAFT
  REVIEW
  PUBLISHED
}

type Services {
  value: String
  label: String
}

type Amenities {
  value: String!
  label: String!
}

type Payments {
  value: String!
  label: String!
}

type Covid {
  value: String!
  label: String!
}

type RestaurantOwner {
  name: String!
  ownerid: String!
  group: String
}

type Sponsors {
  value: String
  label: String
  sponsorName: String
  sponsorIcon: String
}

type OriginalHours {
  open: String
  close: String
}

type Hours {
  open: String
  close: String
  label: String
  open24: Boolean
  closed: Boolean
  original: OriginalHours
  key: String
}

type Days {
  sunday: [Hours]
  monday: [Hours]
  tuesday: [Hours]
  wednesday: [Hours]
  thursday: [Hours]
  friday: [Hours]
  saturday: [Hours]
}

type Location {
  lat: Float
  lon: Float
}

type UserDetails {
  key: String!
  value: String!
}

type BlogCategory {
  value: String
  label: String
}

type Device {
  deviceType: String
  deviceLocation: String
  isCurrentSession: Boolean
}

type Restaurant
  @model
  @searchable
  @aws_api_key
  @aws_iam
  @aws_cognito_user_pools
  @auth(
    rules: [
      { allow: public, operations: [read] }
      { allow: public, provider: iam, operations: [read, create, update] }
      { allow: private, operations: [read, create, update] }
      { allow: owner }
      { allow: owner, ownerField: "owners" }
      { allow: groups, groups: ["admin"] }
      { allow: groups, groups: ["user"], operations: [read, create, update] }
    ]
  ) {
  id: ID!
  owners: [String]
  owner: String
  _status: ModelStatus
  _lastUserToEdit: String
  _lastEdit: String
  _createdAt: String
  name: String
  about: String
  address: String
  route: String
  locality: String
  zip: String!
  phone: String
  website: String
  email: String
  ows: [RestaurantOwner]
  amenities: [Amenities]
  payments: [Payments]
  covid: [Covid]
  details: String
  images: [Images] @hasMany(indexName: "byImages", fields: ["id"])
  location: Location
  placeId: String
  hours: Days
  rating: Float
  services: [Services]
  websiteMenu: String
  sponsors: [Sponsors]
  reviews: [Review] @hasMany(indexName: "byReview", fields: ["id"])
  menu: [Menu] @hasMany(indexName: "byMenu", fields: ["id"])
}

type Images
  @model
  @aws_api_key
  @aws_iam
  @aws_cognito_user_pools
  @auth(
    rules: [
      { allow: public, operations: [read] }
      { allow: public, provider: iam, operations: [read, create, update] }
      { allow: private, operations: [read, create, update] }
      { allow: owner }
      { allow: owner, ownerField: "owners" }
      { allow: groups, groups: ["admin"] }
      { allow: groups, groups: ["user"], operations: [read, create, update] }
    ]
  ) {
  id: ID!
  restaurantid: ID!
    @index(
      name: "listImagesByIndex"
      queryField: "listImagesByIndex"
      sortKeyFields: ["index"]
    )
    @index(name: "byImages")
  _status: ModelStatus
  _lastUserToEdit: String
  _lastEdit: String
  _uploadDate: String
  _createdAt: String
  index: Int
  owner: String
  ownerID: String
  src: String
  restaurant: Restaurant @belongsTo(fields: ["restaurantid"])
}

type Menu
  @model
  @aws_api_key
  @aws_iam
  @aws_cognito_user_pools
  @auth(
    rules: [
      { allow: public, operations: [read] }
      { allow: public, provider: iam, operations: [read, create, update] }
      { allow: private, operations: [read, create, update] }
      { allow: owner }
      { allow: owner, ownerField: "owners" }
      { allow: groups, groups: ["admin"] }
      { allow: groups, groups: ["user"], operations: [read, create, update] }
    ]
  ) {
  id: ID!
  _status: ModelStatus
  _lastUserToEdit: String
  _lastEdit: String
  _createdAt: String
  index: Int
  restaurantid: ID!
    @index(
      name: "listMenuByIndex"
      queryField: "listMenuByIndex"
      sortKeyFields: ["index"]
    )
    @index(name: "byMenu")
  title: String!
  description: String!
  price: String!
  picture: String!
  owner: String!
  restaurant: Restaurant @belongsTo(fields: ["restaurantid"])
}

type Review
  @model
  @aws_api_key
  @aws_iam
  @aws_cognito_user_pools
  @auth(
    rules: [
      { allow: public, operations: [read] }
      { allow: public, provider: iam, operations: [read, create, update] }
      { allow: private, operations: [read, create, update] }
      { allow: owner }
      { allow: owner, ownerField: "owners" }
      { allow: groups, groups: ["admin"] }
      { allow: groups, groups: ["user"], operations: [read, create, update] }
    ]
  ) {
  id: ID!
  _status: ModelStatus
  _lastUserToEdit: String
  _lastEdit: String
  _createdAt: String
  restaurantid: ID! @index(name: "byReview")
  title: String
  description: String
  pictures: [String]
  rating: Float
  userName: String
  userPicture: String
  userEmail: String
  upVotes: Int
  downVotes: Int
  owner: String
  restaurant: Restaurant @belongsTo(fields: ["restaurantid"])
  comments: [ReviewComment]
    @hasMany(indexName: "byReviewComment", fields: ["id"])
}

type ReviewComment
  @model
  @aws_api_key
  @aws_iam
  @aws_cognito_user_pools
  @auth(
    rules: [
      { allow: public, operations: [read] }
      { allow: public, provider: iam, operations: [read, create, update] }
      { allow: private, operations: [read, create, update] }
      { allow: owner }
      { allow: owner, ownerField: "owners" }
      { allow: groups, groups: ["admin"] }
      { allow: groups, groups: ["user"], operations: [read, create, update] }
    ]
  ) {
  id: ID!
  _status: ModelStatus
  _lastUserToEdit: String
  _lastEdit: String
  _createdAt: String
  restaurantid: ID! @index(name: "byReviewComment")
  comment: String
  userName: String
  userPicture: String
  userEmail: String
  upVotes: Int
  downVotes: Int
  owner: String
  restaurantReview: Review @belongsTo(fields: ["restaurantid"])
}

type User
  @model
  @searchable
  @aws_api_key
  @aws_iam
  @aws_cognito_user_pools
  @auth(
    rules: [
      { allow: public, operations: [read] }
      { allow: public, provider: iam, operations: [read, create, update] }
      { allow: private, operations: [read, create, update] }
      { allow: owner }
      { allow: owner, ownerField: "owners" }
      { allow: groups, groups: ["admin"] }
      { allow: groups, groups: ["user"], operations: [read, create, update] }
    ]
  ) {
  id: ID!
  _status: ModelStatus
  _lastUserToEdit: String
  _lastEdit: String
  _createdAt: String
  sub: String!
  name: String
  picture: String
  email: String
  birthdate: String
  phone: String
  bio: String
  personalDataUse: Boolean

  searchEngineSharing: Boolean

  deviceHistory: [Device]

  timezone: String
  preferredLanguage: String
  preferredCurrency: String
}

input QueryLocation {
  lat: Float
  lon: Float
}

type RestaurantConnection @aws_api_key @aws_iam @aws_cognito_user_pools {
  items: [Restaurant]
  total: Int
  nextToken: String
}

type Query {
  exploreRestaurants(
    query: String
    term: String
    total: Int
    nextToken: String
  ): RestaurantConnection @aws_api_key @aws_iam @aws_cognito_user_pools

  boundingBoxRestaurants(
    top_left: QueryLocation
    bottom_right: QueryLocation
    query: String
    fields: [String]
    total: Int
    nextToken: String
  ): RestaurantConnection @aws_api_key @aws_iam @aws_cognito_user_pools
}

```

</details>

### Log output

<details>

```
2022-01-05T09:06:04.590Z|info : amplify-provider-awscloudformation.aws-cfn.getStackEvents.cfnModel.describeStackEvents([{"StackName":"[***]rmation:us-[***]st-[***]30:stack/[***]ify-[***]ox-[***]ev-[***]955/[***]7270-[***]31-[***]ec-[***]dc-[***]36763"}])
2022-01-05T09:06:04.804Z|info : amplify-provider-awscloudformation.aws-cfn.getStackEvents.cfnModel.describeStackEvents([{"StackName":"[***]rmation:us-[***]st-[***]30:stack/[***]ify-[***]ox-[***]ev-[***]955-[***]fox-[***]93EX1-[***]nStack-[***]CD9ZO8/[***]4dc0-[***]3a-[***]ec-[***]b8-[***]437d7"}])
2022-01-05T09:06:04.903Z|info : amplify-provider-awscloudformation.aws-cfn.getStackEvents.cfnModel.describeStackEvents([{"StackName":"[***]rmation:us-[***]st-[***]30:stack/[***]ify-[***]ox-[***]ev-[***]955-[***]fox-[***]93EX1-[***]eStack-[***]75PIC/[***]cd50-[***]3a-[***]ec-[***]d6-[***]43e0d"}])
2022-01-05T09:06:05.003Z|info : amplify-provider-awscloudformation.aws-cfn.getStackEvents.cfnModel.describeStackEvents([{"StackName":"[***]rmation:us-[***]st-[***]30:stack/[***]ify-[***]ox-[***]ev-[***]955-[***]fox-[***]93EX1/[***]9af0-[***]3a-[***]ec-[***]78-[***]0ec17"}])
2022-01-05T09:06:05.601Z|info : amplify-provider-awscloudformation.aws-cfn.getStackEvents.cfnModel.describeStackEvents([{"StackName":"[***]rmation:us-[***]st-[***]30:stack/[***]ify-[***]ox-[***]ev-[***]955-[***]fox-[***]93EX1-[***]nStack-[***]CD9ZO8/[***]4dc0-[***]3a-[***]ec-[***]b8-[***]437d7"}])
2022-01-05T09:06:05.701Z|info : amplify-provider-awscloudformation.aws-cfn.getStackEvents.cfnModel.describeStackEvents([{"StackName":"[***]rmation:us-[***]st-[***]30:stack/[***]ify-[***]ox-[***]ev-[***]955-[***]fox-[***]93EX1-[***]eStack-[***]75PIC/[***]cd50-[***]3a-[***]ec-[***]d6-[***]43e0d"}])
2022-01-05T09:06:09.146Z|info : amplify-provider-awscloudformation.aws-cfn.getStackEvents.cfnModel.describeStackEvents([{"StackName":"[***]ify-[***]ox-[***]ev-[***]955"}])
2022-01-05T09:06:09.466Z|info : amplify-provider-awscloudformation.aws-cfn.getStackEvents.cfnModel.describeStackEvents([{"StackName":"[***]rmation:us-[***]st-[***]30:stack/[***]ify-[***]ox-[***]ev-[***]955-[***]fox-[***]93EX1/[***]9af0-[***]3a-[***]ec-[***]78-[***]0ec17"}])
2022-01-05T09:06:09.565Z|info : amplify-provider-awscloudformation.aws-cfn.getStackEvents.cfnModel.describeStackEvents([{"StackName":"[***]rmation:us-[***]st-[***]30:stack/[***]ify-[***]ox-[***]ev-[***]955/[***]7270-[***]31-[***]ec-[***]dc-[***]36763"}])
2022-01-05T09:06:09.814Z|info : amplify-provider-awscloudformation.aws-cfn.getStackEvents.cfnModel.describeStackEvents([{"StackName":"[***]rmation:us-[***]st-[***]30:stack/[***]ify-[***]ox-[***]ev-[***]955-[***]fox-[***]93EX1-[***]nStack-[***]CD9ZO8/[***]4dc0-[***]3a-[***]ec-[***]b8-[***]437d7"}])
2022-01-05T09:06:09.913Z|info : amplify-provider-awscloudformation.aws-cfn.getStackEvents.cfnModel.describeStackEvents([{"StackName":"[***]rmation:us-[***]st-[***]30:stack/[***]ify-[***]ox-[***]ev-[***]955-[***]fox-[***]93EX1-[***]eStack-[***]75PIC/[***]cd50-[***]3a-[***]ec-[***]d6-[***]43e0d"}])
2022-01-05T09:06:10.014Z|info : amplify-provider-awscloudformation.aws-cfn.getStackEvents.cfnModel.describeStackEvents([{"StackName":"[***]rmation:us-[***]st-[***]30:stack/[***]ify-[***]ox-[***]ev-[***]955-[***]fox-[***]93EX1/[***]9af0-[***]3a-[***]ec-[***]78-[***]0ec17"}])
2022-01-05T09:06:10.325Z|info : amplify-provider-awscloudformation.aws-cfn.getStackEvents.cfnModel.describeStackEvents([{"StackName":"[***]rmation:us-[***]st-[***]30:stack/[***]ify-[***]ox-[***]ev-[***]955-[***]fox-[***]93EX1-[***]nStack-[***]CD9ZO8/[***]4dc0-[***]3a-[***]ec-[***]b8-[***]437d7"}])
2022-01-05T09:06:10.425Z|info : amplify-provider-awscloudformation.aws-cfn.getStackEvents.cfnModel.describeStackEvents([{"StackName":"[***]rmation:us-[***]st-[***]30:stack/[***]ify-[***]ox-[***]ev-[***]955-[***]fox-[***]93EX1-[***]eStack-[***]75PIC/[***]cd50-[***]3a-[***]ec-[***]d6-[***]43e0d"}])
2022-01-05T09:06:14.146Z|info : amplify-provider-awscloudformation.aws-cfn.getStackEvents.cfnModel.describeStackEvents([{"StackName":"[***]ify-[***]ox-[***]ev-[***]955"}])
2022-01-05T09:06:14.468Z|info : amplify-provider-awscloudformation.aws-cfn.getStackEvents.cfnModel.describeStackEvents([{"StackName":"[***]rmation:us-[***]st-[***]30:stack/[***]ify-[***]ox-[***]ev-[***]955-[***]fox-[***]93EX1/[***]9af0-[***]3a-[***]ec-[***]78-[***]0ec17"}])
2022-01-05T09:06:14.567Z|info : amplify-provider-awscloudformation.aws-cfn.getStackEvents.cfnModel.describeStackEvents([{"StackName":"[***]rmation:us-[***]st-[***]30:stack/[***]ify-[***]ox-[***]ev-[***]955/[***]7270-[***]31-[***]ec-[***]dc-[***]36763"}])
2022-01-05T09:06:14.881Z|info : amplify-provider-awscloudformation.aws-cfn.getStackEvents.cfnModel.describeStackEvents([{"StackName":"[***]rmation:us-[***]st-[***]30:stack/[***]ify-[***]ox-[***]ev-[***]955-[***]fox-[***]93EX1/[***]9af0-[***]3a-[***]ec-[***]78-[***]0ec17"}])
2022-01-05T09:06:19.146Z|info : amplify-provider-awscloudformation.aws-cfn.getStackEvents.cfnModel.describeStackEvents([{"StackName":"[***]ify-[***]ox-[***]ev-[***]955"}])
2022-01-05T09:06:19.540Z|info : amplify-provider-awscloudformation.aws-cfn.getStackEvents.cfnModel.describeStackEvents([{"StackName":"[***]rmation:us-[***]st-[***]30:stack/[***]ify-[***]ox-[***]ev-[***]955-[***]fox-[***]93EX1/[***]9af0-[***]3a-[***]ec-[***]78-[***]0ec17"}])
2022-01-05T09:06:19.639Z|info : amplify-provider-awscloudformation.aws-cfn.getStackEvents.cfnModel.describeStackEvents([{"StackName":"[***]rmation:us-[***]st-[***]30:stack/[***]ify-[***]ox-[***]ev-[***]955/[***]7270-[***]31-[***]ec-[***]dc-[***]36763"}])
2022-01-05T09:06:19.964Z|info : amplify-provider-awscloudformation.aws-cfn.getStackEvents.cfnModel.describeStackEvents([{"StackName":"[***]rmation:us-[***]st-[***]30:stack/[***]ify-[***]ox-[***]ev-[***]955-[***]fox-[***]93EX1/[***]9af0-[***]3a-[***]ec-[***]78-[***]0ec17"}])
2022-01-05T09:06:24.146Z|info : amplify-provider-awscloudformation.aws-cfn.getStackEvents.cfnModel.describeStackEvents([{"StackName":"[***]ify-[***]ox-[***]ev-[***]955"}])
2022-01-05T09:06:24.531Z|info : amplify-provider-awscloudformation.aws-cfn.getStackEvents.cfnModel.describeStackEvents([{"StackName":"[***]rmation:us-[***]st-[***]30:stack/[***]ify-[***]ox-[***]ev-[***]955/[***]7270-[***]31-[***]ec-[***]dc-[***]36763"}])
2022-01-05T09:06:29.147Z|info : amplify-provider-awscloudformation.aws-cfn.getStackEvents.cfnModel.describeStackEvents([{"StackName":"[***]ify-[***]ox-[***]ev-[***]955"}])
2022-01-05T09:06:29.482Z|info : amplify-provider-awscloudformation.aws-cfn.getStackEvents.cfnModel.describeStackEvents([{"StackName":"[***]rmation:us-[***]st-[***]30:stack/[***]ify-[***]ox-[***]ev-[***]955/[***]7270-[***]31-[***]ec-[***]dc-[***]36763"}])
2022-01-05T09:06:34.147Z|info : amplify-provider-awscloudformation.aws-cfn.getStackEvents.cfnModel.describeStackEvents([{"StackName":"[***]ify-[***]ox-[***]ev-[***]955"}])
2022-01-05T09:06:34.470Z|info : amplify-provider-awscloudformation.aws-cfn.getStackEvents.cfnModel.describeStackEvents([{"StackName":"[***]rmation:us-[***]st-[***]30:stack/[***]ify-[***]ox-[***]ev-[***]955/[***]7270-[***]31-[***]ec-[***]dc-[***]36763"}])
2022-01-05T09:06:39.147Z|info : amplify-provider-awscloudformation.aws-cfn.getStackEvents.cfnModel.describeStackEvents([{"StackName":"[***]ify-[***]ox-[***]ev-[***]955"}])
2022-01-05T09:06:39.472Z|info : amplify-provider-awscloudformation.aws-cfn.getStackEvents.cfnModel.describeStackEvents([{"StackName":"[***]rmation:us-[***]st-[***]30:stack/[***]ify-[***]ox-[***]ev-[***]955/[***]7270-[***]31-[***]ec-[***]dc-[***]36763"}])
2022-01-05T09:06:44.147Z|info : amplify-provider-awscloudformation.aws-cfn.getStackEvents.cfnModel.describeStackEvents([{"StackName":"[***]ify-[***]ox-[***]ev-[***]955"}])
2022-01-05T09:06:44.505Z|info : amplify-provider-awscloudformation.aws-cfn.getStackEvents.cfnModel.describeStackEvents([{"StackName":"[***]rmation:us-[***]st-[***]30:stack/[***]ify-[***]ox-[***]ev-[***]955/[***]7270-[***]31-[***]ec-[***]dc-[***]36763"}])
2022-01-05T09:06:49.147Z|info : amplify-provider-awscloudformation.aws-cfn.getStackEvents.cfnModel.describeStackEvents([{"StackName":"[***]ify-[***]ox-[***]ev-[***]955"}])
2022-01-05T09:06:49.706Z|info : amplify-provider-awscloudformation.aws-cfn.getStackEvents.cfnModel.describeStackEvents([{"StackName":"[***]rmation:us-[***]st-[***]30:stack/[***]ify-[***]ox-[***]ev-[***]955/[***]7270-[***]31-[***]ec-[***]dc-[***]36763"}])
2022-01-05T09:06:54.147Z|info : amplify-provider-awscloudformation.aws-cfn.getStackEvents.cfnModel.describeStackEvents([{"StackName":"[***]ify-[***]ox-[***]ev-[***]955"}])
2022-01-05T09:06:54.499Z|info : amplify-provider-awscloudformation.aws-cfn.getStackEvents.cfnModel.describeStackEvents([{"StackName":"[***]rmation:us-[***]st-[***]30:stack/[***]ify-[***]ox-[***]ev-[***]955/[***]7270-[***]31-[***]ec-[***]dc-[***]36763"}])
2022-01-05T09:06:59.147Z|info : amplify-provider-awscloudformation.aws-cfn.getStackEvents.cfnModel.describeStackEvents([{"StackName":"[***]ify-[***]ox-[***]ev-[***]955"}])
2022-01-05T09:06:59.758Z|info : amplify-provider-awscloudformation.aws-cfn.getStackEvents.cfnModel.describeStackEvents([{"StackName":"[***]rmation:us-[***]st-[***]30:stack/[***]ify-[***]ox-[***]ev-[***]955/[***]7270-[***]31-[***]ec-[***]dc-[***]36763"}])
2022-01-05T09:07:04.147Z|info : amplify-provider-awscloudformation.aws-cfn.getStackEvents.cfnModel.describeStackEvents([{"StackName":"[***]ify-[***]ox-[***]ev-[***]955"}])
2022-01-05T09:07:04.480Z|info : amplify-provider-awscloudformation.aws-cfn.getStackEvents.cfnModel.describeStackEvents([{"StackName":"[***]rmation:us-[***]st-[***]30:stack/[***]ify-[***]ox-[***]ev-[***]955/[***]7270-[***]31-[***]ec-[***]dc-[***]36763"}])
2022-01-05T09:07:09.147Z|info : amplify-provider-awscloudformation.aws-cfn.getStackEvents.cfnModel.describeStackEvents([{"StackName":"[***]ify-[***]ox-[***]ev-[***]955"}])
2022-01-05T09:07:11.140Z|info : amplify-provider-awscloudformation.aws-cfn.updateamplifyMetaFileWithStackOutputs.cfn.listStackResources([{"StackName":"[***]ify-[***]ox-[***]ev-[***]955"}])
2022-01-05T09:07:11.417Z|info : amplify-provider-awscloudformation.aws-cfn.describeStack.cfn.describeStacks([{"StackName":"[***]ify-[***]ox-[***]ev-[***]955"}])
2022-01-05T09:07:11.642Z|info : amplify-provider-awscloudformation.aws-cfn.describeStack.cfn.describeStacks([{"StackName":"[***]rmation:us-[***]st-[***]30:stack/[***]ify-[***]ox-[***]ev-[***]955-[***]mLambdaStack-[***]WKEAFH/[***]e680-[***]87-[***]ec-[***]eb-[***]90305"}])
2022-01-05T09:07:11.643Z|info : amplify-provider-awscloudformation.aws-cfn.describeStack.cfn.describeStacks([{"StackName":"[***]rmation:us-[***]st-[***]30:stack/[***]ify-[***]ox-[***]ev-[***]955-[***]ueries-[***]SY0HMX/[***]64c0-[***]3a-[***]ec-[***]8d-[***]e38af"}])
2022-01-05T09:07:11.644Z|info : amplify-provider-awscloudformation.aws-cfn.describeStack.cfn.describeStacks([{"StackName":"[***]rmation:us-[***]st-[***]30:stack/[***]ify-[***]ox-[***]ev-[***]955-[***]fox-[***]93EX1/[***]9af0-[***]3a-[***]ec-[***]78-[***]0ec17"}])
2022-01-05T09:07:11.645Z|info : amplify-provider-awscloudformation.aws-cfn.describeStack.cfn.describeStacks([{"StackName":"[***]rmation:us-[***]st-[***]30:stack/[***]ify-[***]ox-[***]ev-[***]955-[***]206fcc-[***]2BP35Y/[***]bd10-[***]3a-[***]ec-[***]58-[***]7b6a7"}])
2022-01-05T09:07:11.645Z|info : amplify-provider-awscloudformation.aws-cfn.describeStack.cfn.describeStacks([{"StackName":"[***]rmation:us-[***]st-[***]30:stack/[***]ify-[***]ox-[***]ev-[***]955-[***]olGroups-[***]EHD2V/[***]2ca0-[***]3a-[***]ec-[***]11-[***]c5e0f"}])
2022-01-05T09:07:11.646Z|info : amplify-provider-awscloudformation.aws-cfn.describeStack.cfn.describeStacks([{"StackName":"[***]rmation:us-[***]st-[***]30:stack/[***]ify-[***]ox-[***]ev-[***]955-[***]xRestaurants-[***]ZGU9U/[***]6340-[***]d0-[***]ec-[***]fe-[***]7eb43"}])
2022-01-05T09:07:11.647Z|info : amplify-provider-awscloudformation.aws-cfn.describeStack.cfn.describeStacks([{"StackName":"[***]rmation:us-[***]st-[***]30:stack/[***]ify-[***]ox-[***]ev-[***]955-[***]estaurants-[***]OOHRR2/[***]5fb0-[***]d0-[***]ec-[***]54-[***]c6b39"}])
2022-01-05T09:07:11.648Z|info : amplify-provider-awscloudformation.aws-cfn.describeStack.cfn.describeStacks([{"StackName":"[***]rmation:us-[***]st-[***]30:stack/[***]ify-[***]ox-[***]ev-[***]955-[***]riesf39e8799-[***]85QU5/[***]5320-[***]3a-[***]ec-[***]e7-[***]b3c41"}])
2022-01-05T09:07:11.650Z|info : amplify-provider-awscloudformation.aws-cfn.describeStack.cfn.describeStacks([{"StackName":"[***]rmation:us-[***]st-[***]30:stack/[***]ify-[***]ox-[***]ev-[***]955-[***]stConfirmation-[***]201DCC/[***]a400-[***]3a-[***]ec-[***]e7-[***]b3c41"}])
2022-01-05T09:07:11.650Z|info : amplify-provider-awscloudformation.aws-cfn.describeStack.cfn.describeStacks([{"StackName":"[***]rmation:us-[***]st-[***]30:stack/[***]ify-[***]ox-[***]ev-[***]955-[***]Storage-[***]UA06E/[***]a830-[***]3a-[***]ec-[***]11-[***]e789f"}])
2022-01-05T09:07:13.016Z|info : amplify-provider-awscloudformation.push-resources.storeCurrentcoudBackend.s3.uploadFile([{"Key":"[***]rent-[***]ud-[***]d.zip"}])
2022-01-05T09:07:13.018Z|info : amplify-provider-awscloudformation.aws-s3.uploadFile.s3.putObject([{"Key":"[***]rent-[***]ud-[***]d.zip","Bucket":"[***]ify-[***]ox-[***]ev-[***]955-[***]ment"}])
2022-01-05T09:07:13.643Z|info : amplify-provider-awscloudformation.amplify-service-manager.s3.uploadFile([{"Key":"[***]ify-[***]json"}])
2022-01-05T09:07:13.644Z|info : amplify-provider-awscloudformation.aws-s3.uploadFile.s3.putObject([{"Key":"[***]ify-[***]json","Bucket":"[***]ify-[***]ox-[***]ev-[***]955-[***]ment"}])
2022-01-05T09:07:13.879Z|info : amplify-provider-awscloudformation.amplify-service-manager.s3.uploadFile([{"Key":"[***]end-[***].json"}])
2022-01-05T09:07:13.879Z|info : amplify-provider-awscloudformation.aws-s3.uploadFile.s3.putObject([{"Key":"[***]end-[***].json","Bucket":"[***]ify-[***]ox-[***]ev-[***]955-[***]ment"}])
```

</details>

**Comment by Rafcin** [comment]
The searchable issue it seems is caused by running _reindex and changing the template structure. In my project one of my fields named `location` takes `lat, lng` normally, but for the search queries to work it have to map it as `geo_point`.

Using reindex and having 2 sources works, however after reindex is used the searchable query fails I assume because the mapping is different. Is there any way I can permanently tell OS to map location as `geo_point` and update that info for the searchable queries to work?

This is my current method in OS

```
#Make a new doc for fixed types
PUT /fixed_mappings_for_restaurant
{
  "mappings": {
    "properties": {
      "location": {
        "type": "geo_point"
      }
    }
  }
}

#Move old data to new doc
POST _reindex
{
   "source":{
      "index":"restaurant"
   },
   "dest":{
      "index":"fixed_mappings_for_restaurant"
   }
}

#delete old doc
DELETE /restaurant

#Make a new doc with the original name for fixed types
PUT /restaurant
{
  "mappings": {
    "properties": {
      "location": {
        "type": "geo_point"
      }
    }
  }
}

#Move new data back to old doc with fixed mapping
POST _reindex
{
   "source":{
      "index":"fixed_mappings_for_restaurant"
   },
   "dest":{
      "index":"restaurant"
   }
}
```

**Comment by Rafcin** [comment]
https://docs.amplify.aws/cli-legacy/graphql-transformer/resolvers/#add-a-custom-geolocation-search-resolver-that-targets-an-opensearch-domain-created-by-searchable
**Side Note**
If they legacy docs are still to be trusted, if you are using OpenSearch and follow the geo_point example. The res.vtl total has changed its path to `$ctx.result.hits.total.value` instead of `$ctx.result.hits.total`

**Original**
```
## Query.nearbyTodos.res.vtl

#set( $items = [] )
#foreach( $entry in $context.result.hits.hits )
  #if( !$foreach.hasNext )
    #set( $nextToken = "$entry.sort.get(0)" )
  #end
  $util.qr($items.add($entry.get("_source")))
#end
$util.toJson({
  "items": $items,
  "total": $ctx.result.hits.total,
  "nextToken": $nextToken
})
```

**OpenSearch**
```
#set( $items = [] )
#foreach( $entry in $context.result.hits.hits )
    #if( !$foreach.hasNext )
        #set( $nextToken = "$entry.sort.get(0)" )
    #end
    $util.qr($items.add($entry.get("_source")))
#end
$util.toJson({
"items": $items,
  "total": $ctx.result.hits.total.value,
  "nextToken": $nextToken
})
```
The index path can also drop /doc `#set( $indexPath = "/todo/doc/_search" )` to `#set( $indexPath = "/todo/_search" )`

**Comment by Rafcin** [comment]
Yea, looks like if you reindex a doc and change the type the autogenerate searchable queries will no longer function properly.

**Comment by edwardfoyle** [comment]
@Rafcin The new GraphQL transformer uses OS 7.x instead of 6.x. I believe the issues you are seeing are a result of that version bump. Bumping this version can cause the OS cluster to lose data which may explain why you aren't seeing any searchable results. We recommend taking a snapshot of your OS cluster before migration and then restoring the snapshot after migration (see: https://docs.amplify.aws/cli/migration/transformer-migration/#changes-that-amplify-cli-requires-you-to-manually-migrate)

The legacy docs should only be referenced when using V1 of the GraphQL Transformer and do not necessarily apply to the V2 Transformer.

Regarding the issue you're having with IAM authorizers, I'll dig into that a bit more and get back to you.

**Comment by Rafcin** [comment]
@edwardfoyle I managed to resolve the issue and I put a note in the migration docs. It took me a moment to figure it out, it OS 1.1 no longer needs `/doc` and the Appsync searchable resolvers that are set up via the transformer still used /doc/_search causing it to break. In my note, I added that a client can use the vtl override and just copy the resolver and change the index path. That worked fine for me. Everything else after that worked fine. 

I think a check should be added in the searchable transformer to update the index path if the user is using >= 1.1 although I have no idea how to get the version info of the deployed OS instance to verify this, hence the vtl override workaround.

The Appsync IAM issue ends up being a problem in the Appsync console, I was able to make calls from my app. A user on Discord informed me that was the case. The permissions matrix verified this for me as well.
